packages/plugin-commands-audit/src/doctor.ts

```diff
@@ -0,0 +1,62 @@
+import renderHelp from 'render-help'
+import { docsUrl } from '@pnpm/cli-utils'
+import { logger } from '@pnpm/logger'
+
+export const rcOptionsTypes = cliOptionsTypes
+
+export function cliOptionsTypes () {
+  return {
+    config: Boolean,
+  }
+}
+
+export const shorthands = {
+  C: '--config',
+}
+
+export const commandNames = ['doctor']
+
+export function help () {
+  return renderHelp({
+    description: 'Checks for known common issues.',
+    descriptionLists: [
+      {
+        title: 'Options',
+        list: [
+          {
+            description: 'Check the global config',
+            name: '--config',
+            shortAlias: '-C',
+          },
+        ],
+      },
+    ],
+    url: docsUrl('doctor'),
+    usages: ['pnpm doctor [options]'],
+  })
+}
+
+export async function handler (
+  opts: {
+    config: boolean
+  }) {
+  if (opts.config) {
+    const paths = require.resolve.paths('npm')
+    try {
+      require.resolve('npm', { paths: paths?.slice(-1) })
```

Comment:
This is already done in the `@pnpm/npm-conf` package. So I guess npm-conf can set a setting when this fails. Like `failedToLoadBuiltInConfig`. And that setting will be of course passed to this plugin throw options.

Reply:
It's done in https://github.com/pnpm/npm-conf/pull/8